### PR TITLE
added transaction module with durable decorator

### PR DESCRIPTION
**Pull request by thinkt4nk**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -4,7 +4,6 @@ envlist =
     py{36}-django20
     py{36,37}-django21
     py{36,37}-django22
-    py{36,37}-djangomaster
```
Comment (somewes) [inline tox.ini:7]: why was this removed?

Comment (somewes) [reply tox.ini:7]: nevermind, looks like this doesn't work anymore

File: ambition_utils/version.py [hunk 2]
```diff
@@ -1 +1 @@
-__version__ = '2.0.0'
+__version__ = '2.1.0'
```
Comment (somewes) [inline ambition_utils/version.py:1]: you'll have to bump this and your notes to 2.2.0

**Comment by thinkt4nk** [comment]
@somewes should be good to go now

**Comment by somewes** [comment]
@thinkt4nk can you add a test that uses a test model + fake external api call to create a somewhat real scenario that demonstrates how this will be used? it would be good to have a test showing

atomic
    atomic code
    bad code

followed by the
durable
    atomic code
    bad code

case like mentioned in the article. i'd rather have these cases set up like this rather than mocking the transaction calls.

i see it says you can't wrap durable calls, but what happens if you try to use durable inside a regular atomic transaction?

can't wait to upgrade to django 3.2 since this option is built in.

**Comment by thinkt4nk** [comment]
@somewes well that's pretty much what I did. I didn't mock atomic

**Comment by thinkt4nk** [comment]
@somewes looking at this again this morning, the only one I mocked was for coverage, but the other two test the two important cases without mocking. If you want to jump on a call to discuss, I'm available all morning

**Comment by somewes** [comment]
when i said mocking the transaction calls, i meant the `get_autocommit` calls. i see that you are hitting the code, but what i was asking for is more of an integration test. but there probably isn't really a case where we would be outside of an atomic block and have auto commit turned off, so that is fine.
